networkit/cpp/community/OverlappingNMIDistance.cpp

```diff
@@ -0,0 +1,393 @@
+// networkit-format
+
+#include <cmath>
+#include <iomanip>
+#include <numeric>
+#include <unordered_map>
+
+#include <networkit/auxiliary/NumericTools.hpp>
+#include <networkit/auxiliary/SignalHandling.hpp>
+#include <networkit/community/OverlappingNMIDistance.hpp>
+
+namespace NetworKit {
+
+namespace {
+
+struct pair_hash {
+    template <class T>
+    static inline void hash_combine(std::size_t &seed, T const &v) {
+        // https://www.boost.org/doc/libs/1_75_0/doc/html/hash/reference.html#boost.hash_combine
+        seed ^= std::hash<T>()(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+    }
+
+public:
+    std::size_t operator()(const std::pair<index, index> &pair) const {
+        std::size_t seed = 0;
+        hash_combine(seed, pair.first);
+        hash_combine(seed, pair.second);
+        return seed;
+    }
+};
```

Comment:
This might be useful also in other parts of NetworKit, it's probably a good idea to move it to the `auxiliary` module.

Reply:
That seems like a good idea. I added `Aux::hashCombine` and `Aux::PairHash` in a `HashUtils.hpp` (e9590cd).